base/reflection.jl

```diff
@@ -2342,3 +2342,7 @@ function destructure_callex(topmod::Module, @nospecialize(ex))
     end
     return f, args, kwargs
 end
+
+generating_image() = ccall(:jl_generating_output, Cint, ()) != 0
+
+generating_system_image() = generating_image() && JLOptions().incremental == 0
```

Comment:
I feel like this should an optional boolean argument, instead of 2 functions, like so
```suggestion
"""
    Base.generating_output([incremental::Bool])::Bool

Return `true` if the current process is being used to pre-generate a
code cache via any of the `--output-*` command line arguments. The
optional argument `incremental` controls whether to only return `true`
for that specific mode of compilation.
"""
generating_output(incremental::Union{Bool,Nothing}=nothing) =
    ccall(:jl_generating_output, Cint, ()) != 0 && !JLOptions().incremental !== incremental
```